lib/runner/testsuite.js

```diff
@@ -203,20 +208,21 @@ TestSuite.prototype.run = function() {
     this.globalBeforeEach()
       .then(function() {
         if (self.client.terminated() && self.client.skipTestcasesOnFail()) {
-          self.testResults.errmessages = self.client.errors();
-          self.deferred.resolve(self.testResults);
           return null;
         }
-        return self.runTestSuiteModule();
-      })
 
-      .then(function(results) {
-        return self.globalAfterEach();
+        return self.runTestSuiteModule()
+          .then(function(results) {
+            return self.globalAfterEach();
+          });
       })
 
```

Comment:
Need to investigate this promise flow a little more deeper when there are errors.